### PR TITLE
TTAHUB-1721 Trim Goals and Objectives prior to creating in db

### PR DESCRIPTION
**Pull request by kcirtapfromspace**
## Description of change
Apply whitespace trim during goal creation to prevent duplicate goal templates from being created due to prefix and suffix adds.  

## How to test
Create a goal with excess whitespace. From both AR Goals & recipient-tta. Check db for trimmed goals

```
     <---There are 5 spaces--->     
```

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1721

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/services/goals.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ const OPTIONS_FOR_GOAL_FORM_QUERY = (id, recipientId) => ({
   attributes: [
     'id',
     'endDate',
-    'name',
```
Comment (GarrettEHill) [inline src/services/goals.js:53]: this wont work, that is building the command for sequelize, it does not work like that.

File: src/services/goals.js [hunk 2]
```diff
@@ -72,7 +72,7 @@ const OPTIONS_FOR_GOAL_FORM_QUERY = (id, recipientId) => ({
   include: [
     {
       attributes: [
-        'title',
+        'title'.trim(),
```
Comment (GarrettEHill) [inline src/services/goals.js:75]: this will not work

File: src/services/goals.js (hunk 1 above)
Comment (GarrettEHill) [reply src/services/goals.js:53]: please remove all the ''.trim()

Comment (kcirtapfromspace) [reply src/services/goals.js:53]: 👍

**Review by GarrettEHill — APPROVED** [review]

**Comment by kcirtapfromspace** [comment]
> We need to add trim to one additional place, about line 1878 in services/goal, see below. A great way to validate is to update the goal name createOrUpdateGoalsForActivityReport.test.js:316 with a bunch of extra spaces and make sure the test passes.
> 
> ```js
>     // if not, we create it
>       if (!newOrUpdatedGoal) {
>         newOrUpdatedGoal = await Goal.create({
>           createdVia: 'activityReport',
>           grantId,
>           ...fields,
>           name: fields.name.trim(),
>           status,
>         }, { individualHooks: true });
>       }
> ```

Added some spaces and the trim in the ActivityReport create goal 
 PASS  src/services/groups.test.js (1044 MB heap size)

**Pull request merged**